Platform only descriptors not possible

**Describe the bug**
The docs state that using a platform-only descriptor should be possible.
https://julieops.readthedocs.io/en/latest/futures/what-acl-management.html?highlight=platform#cluster-wide-roles

```yml
---
  context: "context"
  platform:
      kafka:
        rbac:
          SecurityAdmin:
            - principal: "User:Foo"
          ClusterAdmin:
            - principal: "User:Boo"
      kafka_connect:
        rbac:
          SecurityAdmin:
            - principal: "User:Foo"
      schema_registry:
        instances:
          - principal: "User:SchemaRegistry01"
            topic: "foo"
            group: "bar"
          - principal: "User:SchemaRegistry02"
            topic: "zet"
        rbac:
          Operator:
            - principal: "User:Hans"
            - principal: "User:Bob"
```
It results in the following error:

```
Caused by: java.io.IOException: projects is a required field in the topology, please specify.
```

**Runtime (please complete the following information):**
 - OS: CentOS7
- JVM version: openjdk 1.8.0
 - Version 2.1.2

--- a/src/main/java/com/purbon/kafka/topology/serdes/TopologyCustomDeserializer.java
+++ b/src/main/java/com/purbon/kafka/topology/serdes/TopologyCustomDeserializer.java
@@ -104,10 +104,10 @@ public class TopologyCustomDeserializer extends StdDeserializer<Topology> {
     validateRequiresKeys(rootNode, CONTEXT_KEY);
     if (rootNode.get(PROJECTS_KEY) == null) {
       LOGGER.warn(
-              PROJECTS_KEY
-                      + " is missing for topology: "
-                      + rootNode.get(CONTEXT_KEY).asText()
-                      + ", this might be a required field, be aware.");
+          PROJECTS_KEY
+              + " is missing for topology: "
+              + rootNode.get(CONTEXT_KEY).asText()
+              + ", this might be a required field, be aware.");
     }
 
     Topology topology = new TopologyImpl(config);
@@ -142,7 +142,7 @@ public class TopologyCustomDeserializer extends StdDeserializer<Topology> {
     topology.setPlatform(platform);
     if (rootNode.get(PROJECTS_KEY) != null) {
       parseProjects(parser, rootNode.get(PROJECTS_KEY), topology, config)
-              .forEach(topology::addProject);
+          .forEach(topology::addProject);
 
       // validate the generated full topics names for valid encoding
       for (Project project : topology.getProjects()) {

--- a/src/main/java/com/purbon/kafka/topology/serdes/TopologyCustomDeserializer.java
+++ b/src/main/java/com/purbon/kafka/topology/serdes/TopologyCustomDeserializer.java
@@ -101,7 +101,14 @@ public class TopologyCustomDeserializer extends StdDeserializer<Topology> {
       throws IOException {
 
     JsonNode rootNode = parser.getCodec().readTree(parser);
-    validateRequiresKeys(rootNode, CONTEXT_KEY, PROJECTS_KEY);
+    validateRequiresKeys(rootNode, CONTEXT_KEY);
+    if (rootNode.get(PROJECTS_KEY) == null) {
+      LOGGER.warn(
+              PROJECTS_KEY
+                      + " is missing for topology: "
+                      + rootNode.get(CONTEXT_KEY).asText()
+                      + ", this might be a required field, be aware.");
+    }
 
     Topology topology = new TopologyImpl(config);
     List<String> excludeAttributes = Arrays.asList(PROJECTS_KEY, CONTEXT_KEY, PLATFORM_KEY);
@@ -133,13 +140,15 @@ public class TopologyCustomDeserializer extends StdDeserializer<Topology> {
     }
 
     topology.setPlatform(platform);
-    parseProjects(parser, rootNode.get(PROJECTS_KEY), topology, config)
-        .forEach(topology::addProject);
+    if (rootNode.get(PROJECTS_KEY) != null) {
+      parseProjects(parser, rootNode.get(PROJECTS_KEY), topology, config)
+              .forEach(topology::addProject);
 
-    // validate the generated full topics names for valid encoding
-    for (Project project : topology.getProjects()) {
-      for (Topic topic : project.getTopics()) {
-        validateEncodingForTopicName(topic.toString());
+      // validate the generated full topics names for valid encoding
+      for (Project project : topology.getProjects()) {
+        for (Topic topic : project.getTopics()) {
+          validateEncodingForTopicName(topic.toString());
+        }
       }
     }
 

--- a/src/test/java/com/purbon/kafka/topology/TopologyObjectBuilderTest.java
+++ b/src/test/java/com/purbon/kafka/topology/TopologyObjectBuilderTest.java
@@ -30,7 +30,6 @@ public class TopologyObjectBuilderTest {
 
   @Test
   public void buildOnlyConnectorTopo() throws IOException {
-
     HashMap<String, String> cliOps = new HashMap<>();
     cliOps.put(BROKERS_OPTION, "");
     Properties props = new Properties();
@@ -42,6 +41,15 @@ public class TopologyObjectBuilderTest {
 
     var proj = topology.getProjects().get(0);
     assertThat(proj.getConnectorArtefacts().getConnectors()).hasSize(2);
+  }
+
+  @Test
+  public void buildTopoOnlyPlatform() throws IOException {
+    String fileOrDirPath = TestUtils.getResourceFilename("/descriptor-only-platform.yaml");
+    Topology topology = TopologyObjectBuilder.build(fileOrDirPath);
+    assertThat(topology.getProjects()).hasSize(0);
+    assertThat(topology.getPlatform().getKafka().getRbac()).isPresent();
+    assertThat(topology.getPlatform().getKafka().getRbac().get()).hasSize(2);
   }
 
   @Test

--- a/src/test/java/com/purbon/kafka/topology/TopologySerdesTest.java
+++ b/src/test/java/com/purbon/kafka/topology/TopologySerdesTest.java
@@ -195,8 +195,8 @@ public class TopologySerdesTest {
     parser.deserialise(TestUtils.getResourceFile("/descriptor-with-no-context.yaml"));
   }
 
-  @Test(expected = TopologyParsingException.class)
-  public void testTopologyWithNoProject() {
+  @Test
+  public void testTopologyWithNoProjectShouldNotError() {
     parser.deserialise(TestUtils.getResourceFile("/descriptor-with-no-project.yaml"));
   }
 

--- a/src/test/resources/descriptor-only-platform.yaml
+++ b/src/test/resources/descriptor-only-platform.yaml
@@ -1,0 +1,9 @@
+---
+context: "context"
+platform:
+  kafka:
+    rbac:
+      SecurityAdmin:
+        - principal: "User:Foo"
+      ClusterAdmin:
+        - principal: "User:Boo"

